File: cpp/204 - Count Primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if (n <= 2)
            return 0;
            
        // var
        int count = 1;
        vector<bool> prime(n, true);
        int m = prime.size();
        
        // init
        for (long i = 3; i < n; i += 2) {
            if (prime[i]) {
                for (long j = i * i; j < n; j += i)
                    prime[j] = false;
                count++;
            }
        }
        return count;
    }
};

File: python/204 - Count Primes.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n < 2:
            return 0
        
        prime = [True] * n
        prime[0] = False
        prime[1] = False
        
        i = 2
        count = 0
        while i < n:
            if prime[i]:
                count += 1
                j = 2 * i
                while j < n:
                    prime[j] = False
                    j += i
            i += 1

        return count